阿里巴巴dubbo2.6.12+ seata-all 1.8.0 报错

14:05:33.127 [localhost-startStop-1] ERROR [] o.s.web.context.ContextLoader Line:350 - Context initialization failed
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'com.easyhome.cloudservice.order.service.IIntegralMallReturnOrderService': Initialization of bean failed; nested exception is java.lang.RuntimeException: io.seata.common.exception.FrameworkException: method not found: Method<ServiceBean.getInterfaceClass()>
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:564) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:483) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:761) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:866) ~[spring-context-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:542) ~[spring-context-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:443) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:325) ~[spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:107) [spring-web-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:5182) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5705) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:1018) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:994) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:652) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:819) [catalina.jar:7.0.92.2568]
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:2119) [catalina.jar:7.0.92.2568]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_242]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_242]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_242]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_242]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_242]
Caused by: java.lang.RuntimeException: io.seata.common.exception.FrameworkException: method not found: Method<ServiceBean.getInterfaceClass()>
	at io.seata.spring.annotation.GlobalTransactionScanner.wrapIfNecessary(GlobalTransactionScanner.java:325) ~[seata-all-1.8.0.jar:1.8.0]
	at org.springframework.aop.framework.autoproxy.AbstractAutoProxyCreator.postProcessAfterInitialization(AbstractAutoProxyCreator.java:298) ~[spring-aop-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsAfterInitialization(AbstractAutowireCapableBeanFactory.java:423) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1633) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555) ~[spring-beans-4.3.7.RELEASE.jar:4.3.7.RELEASE]
	... 24 common frames omitted
Caused by: io.seata.common.exception.FrameworkException: method not found: Method<ServiceBean.getInterfaceClass()>
	at io.seata.rm.tcc.remoting.parser.DubboRemotingParser.getServiceDesc(DubboRemotingParser.java:66) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.rm.tcc.remoting.parser.DefaultRemotingParser.parserRemotingServiceInfo(DefaultRemotingParser.java:167) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.spring.util.TCCBeanParserUtils.parserRemotingServiceInfo(TCCBeanParserUtils.java:179) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.spring.util.TCCBeanParserUtils.isTccAutoProxy(TCCBeanParserUtils.java:50) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.spring.annotation.GlobalTransactionScanner.wrapIfNecessary(GlobalTransactionScanner.java:283) ~[seata-all-1.8.0.jar:1.8.0]
	... 28 common frames omitted
Caused by: java.lang.NoSuchMethodException: method not found: Method<ServiceBean.getInterfaceClass()>
	at io.seata.common.util.ReflectionUtil.getMethod(ReflectionUtil.java:465) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.common.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:564) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.common.util.ReflectionUtil.invokeMethod(ReflectionUtil.java:583) ~[seata-all-1.8.0.jar:1.8.0]
	at io.seata.rm.tcc.remoting.parser.DubboRemotingParser.getServiceDesc(DubboRemotingParser.java:51) ~[seata-all-1.8.0.jar:1.8.0]
	... 32 common frames omitted

--- a/changes/en-us/develop.md
+++ b/changes/en-us/develop.md
@@ -12,6 +12,7 @@ Add changes here for all PR submitted to the develop branch.
 - [[#6026](https://github.com/seata/seata/pull/6026)] fix incorrect metric report
 - [[#4410](https://github.com/seata/seata/pull/4410)] support jdk9+ compile code
 - [[#6104](https://github.com/seata/seata/pull/6104)] fix dubbo 3.x consumer can't generate TCC proxy in tcc mode.
+- [[#6409](https://github.com/seata/seata/pull/6409)] fix the failure of RemotingParser to prevent AT mode from executing.
 
 ### optimize:
 - [[#6044](https://github.com/seata/seata/pull/6044)] optimize derivative product check base on mysql

--- a/changes/zh-cn/develop.md
+++ b/changes/zh-cn/develop.md
@@ -12,6 +12,7 @@
 - [[#6026](https://github.com/seata/seata/pull/6026)] 修复异常的打点
 - [[#4410](https://github.com/seata/seata/pull/4410)] 修复jdk9+版本编译后，引入后ByteBuffer#flip NoSuchMethodError的问题
 - [[#6104](https://github.com/seata/seata/pull/6104)] 修复在TCC模式下, dubbo 3.x版本消费者端不能生成TCC代理的问题
+- [[#6409](https://github.com/seata/seata/pull/6409)] 修复 RemotingParser 失败导致 AT 模式无法执行的问题
 
 ### optimize:
 - [[#6044](https://github.com/seata/seata/pull/6044)] 优化MySQL衍生数据库判断逻辑

--- a/tcc/src/main/java/io/seata/rm/tcc/remoting/parser/DefaultRemotingParser.java
+++ b/tcc/src/main/java/io/seata/rm/tcc/remoting/parser/DefaultRemotingParser.java
@@ -164,7 +164,12 @@ public class DefaultRemotingParser {
      * @return remoting desc
      */
     public RemotingDesc parserRemotingServiceInfo(Object bean, String beanName, RemotingParser remotingParser) {
-        RemotingDesc remotingBeanDesc = remotingParser.getServiceDesc(bean, beanName);
+        RemotingDesc remotingBeanDesc = null;
+        try {
+            remotingBeanDesc = remotingParser.getServiceDesc(bean, beanName);
+        } catch (Throwable ignore) {
+            //ignore exception. It's not possible to judge whether the TCC mode was used.
+        }
         if (remotingBeanDesc == null) {
             return null;
         }

